### PR TITLE
Replace scheduler

### PR DESCRIPTION
**Pull request by NKcqx**
Sync to the newest internal code.
Module compile state:
src/main

- [x] streaming-api
- [x] streaming-state
- [x] streaming-runtime
- [x] streaming-common

src/test

- [x] streaming-api
- [x] streaming-state
- [x] streaming-runtime
- [x] streaming-common

Module UT:

- [x] streaming-api: 100%
- [x] streaming-state: 100%
- [ ] streaming-runtime: 70%
- [x] streaming-common: 100%

### CODE REVIEW

File: streaming/java/streaming-api/src/main/java/io/ray/streaming/api/collector/CollectionCollector.java [hunk 1]
```diff
@@ -10,16 +10,25 @@
...
-      collector.collect(new Record(value));
+    for (Collector<Record<T>> collector : collectorList) {
+      collector.collect(new Record<>(value));
     }
   }
-}
+//
+//  @Override
+//  public void retract(T value) {
+//    for (Collector<Record<T>> collector : collectorList) {
```
Comment (BalaBalaYi) [inline streaming/java/streaming-api/src/main/java/io/ray/streaming/api/collector/CollectionCollector.java:28]: remove unused code

File: streaming/java/streaming-api/src/main/java/io/ray/streaming/jobgraph/VertexType.java [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 /** Different roles for a node. */
 public enum VertexType {
-  SOURCE,
-  TRANSFORMATION,
-  SINK,
+  source,
```
Comment (BalaBalaYi) [inline streaming/java/streaming-api/src/main/java/io/ray/streaming/jobgraph/VertexType.java:5]: need some annotations for enum

File: streaming/java/streaming-api/src/main/java/io/ray/streaming/operator/AbstractStreamOperator.java [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+      LOG.debug("collect null, key: {}, value: {}", key, value);
+      return;
+    }
+    for (Collector collector : this.collectorList) {
+      collector.collect(new KeyRecord(key, value));
+    }
+  }
+//
+//  protected <T> void retract(T record) {
+//    for (Collector collector : this.collectorList) {
```
Comment (BalaBalaYi) [inline streaming/java/streaming-api/src/main/java/io/ray/streaming/operator/AbstractStreamOperator.java:152]: unused code

File: streaming/java/streaming-api/src/main/java/io/ray/streaming/operator/chain/ForwardCollector.java [hunk 4]
```diff
@@ -4,21 +4,44 @@
...
     }
   }
+
+//  @Override
+//  public void retract(T value) {
+//    Record record;
+//    if (value instanceof Record) {
+//      record = (Record) value;
+//    } else {
+//      record = new Record(value);
```
Comment (BalaBalaYi) [inline streaming/java/streaming-api/src/main/java/io/ray/streaming/operator/chain/ForwardCollector.java:37]: we can add a default impl in abstraction if the retraction is not supported yet

File: streaming/java/streaming-state/src/test/java/io/ray/streaming/state/impl/DefaultKeyValueStoreSerializationTest.java [hunk 5]
```diff
@@ -1,44 +0,0 @@
...
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package io.ray.streaming.state.impl;
-
-import io.ray.streaming.state.serialization.impl.DefaultKeyValueStoreSerialization;
```
Comment (BalaBalaYi) [inline streaming/java/streaming-state/src/test/java/io/ray/streaming/state/impl/DefaultKeyValueStoreSerializationTest.java:21]: why these test case deleted?

File: streaming/java/streaming-state/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+## 各模块功能 & 关系
+* keystate：_streaming-state_ 的 api，其内容包括：
```
Comment (BalaBalaYi) [inline streaming/java/streaming-state/README.md:2]: need another readme in english version

File: streaming/java/streaming-runtime/pom_template.xml [hunk 7]
```diff
@@ -46,12 +47,33 @@
       <version>${ray.version}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>io.ant-ray</groupId>
+      <artifactId>common-libs-actor-optimizes</artifactId>
```
Comment (BalaBalaYi) [inline streaming/java/streaming-runtime/pom_template.xml:52]: remove this one (it's used for cougar, useless for open source)

File: streaming/java/streaming-runtime/pom_template.xml [hunk 8]
```diff
@@ -46,12 +47,33 @@
...
     <dependency>
       <groupId>io.ray</groupId>
       <artifactId>streaming-api</artifactId>
       <version>${streaming.version}</version>
     </dependency>
-    {generated_bzl_deps}
+    <dependency>
+      <groupId>io.ray</groupId>
+      <artifactId>streaming-common</artifactId>
+      <version>${streaming.version}</version>
```
Comment (BalaBalaYi) [inline streaming/java/streaming-runtime/pom_template.xml:69]: streaming-api depends on streaming-common, so streaming-runtime no need to depend on streaming-common directly

File: streaming/java/streaming-common/src/main/java/io/ray/streaming/common/utils/EnvUtil.java [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+      LOG.debug("{} env variable is emtpy.", NAMESPACE_ENV_KEY);
+    }
+    return namespace;
+  }
+
+  public static String getRuntimeEnv(String envName) {
+    return System.getenv(envName);
+  }
+
+  public static boolean isOnlineEnv() {
```
Comment (BalaBalaYi) [inline streaming/java/streaming-common/src/main/java/io/ray/streaming/common/utils/EnvUtil.java:54]: should be removed

File: streaming/java/streaming-common/src/main/java/io/ray/streaming/common/utils/EnvUtil.java [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EnvUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(EnvUtil.class);
+
+  public static final String DEV_ENV_VALUE = "dev";
```
Comment (BalaBalaYi) [inline streaming/java/streaming-common/src/main/java/io/ray/streaming/common/utils/EnvUtil.java:25]: also can be removed

Comment (NKcqx) [reply streaming/java/streaming-common/src/main/java/io/ray/streaming/common/utils/EnvUtil.java:25]: Then use what cluster name if started as local mode, this name is needed to build `WorkerRuntimeInfo`

File: streaming/java/streaming-state/src/test/java/io/ray/streaming/state/impl/DefaultKeyValueStoreSerializationTest.java (hunk 5 above)
Comment (NKcqx) [reply streaming/java/streaming-state/src/test/java/io/ray/streaming/state/impl/DefaultKeyValueStoreSerializationTest.java:21]: Haven't yet replace with the newest UT

**Review by BalaBalaYi — APPROVED** [review]
Merge all the code changing 1st. I will solve the cpp+java dependency problem later.

**Pull request merged**